runhouse/resources/hardware/cluster.py

```diff
@@ -180,19 +181,65 @@ def from_name(
         alt_options=None,
         _resolve_children=True,
     ):
-        cluster = super().from_name(
-            name=name,
-            dryrun=dryrun,
-            alt_options=alt_options,
-            _resolve_children=_resolve_children,
-        )
+        try:
+            cluster = super().from_name(
+                name=name,
+                load_from_den=load_from_den,
+                dryrun=dryrun,
+                alt_options=alt_options,
+                _resolve_children=_resolve_children,
+            )
+        except ValueError as e:
+            if load_from_den:
+                # not found in Den
+                raise e
+
+            # Try re-loading the cluster with the Sky DB + cluster config
+            cluster = cls._from_sky_db(name, dryrun)
+
         if hasattr(cluster, "_update_from_sky_status"):
             try:
                 cluster._update_from_sky_status(dryrun=True)
             except:
                 pass
+
         return cluster
 
+    @classmethod
+    def _from_sky_db(cls, name, dryrun=False):
+        import sky
+
+        from runhouse import OnDemandCluster
+
+        sky_cluster_name = os.path.basename(name)
+        sky_data: dict = sky.global_user_state.get_cluster_from_name(sky_cluster_name)
+        if not sky_data:
+            raise ValueError(f"Cluster {name} not found locally")
+
+        cluster_ip = sky_data["handle"].head_ip
+        if not cluster_ip:
+            raise ValueError(
+                f"Failed to load cluster {name} locally, IP address not found"
+            )
+
+        try:
+            resp = requests.get(
```

Comment:
do you know when this request succeeds/fails?

just to clarify expected behavior - wondering if this request fails but a sky cluster is detected whether or not we should go ahead and create it anyways or if it should fail (or something in between, maybe checking that runhouse server is up?). currently in main we have this branch where if from_name fails but the cluster is detected locally it's loosely reconstructed (set properties like provider/instance type may be lost, but I'm working on adding observed properties to the config now).

https://github.com/run-house/runhouse/pull/1093/files#diff-e3919e7abe1f5cd55b08e55638e03b6935258ed6dd0c50afc4f419cb2a59b554R467

Reply:
oh nice, actually this logic should prob be called within that block that you recently added for checking the state. 

i set a timeout on the request, but it can fail for a number of reasons (daemon is down, port isn't open, etc.). I feel like us surfacing a more relevant exception is more appropriate than re-upping the cluster entirely in this case

Reply:
I think I tried upping a local only cluster and the request timed out (but the runhouse server was up) so I wasn't sure why that was happening. so if that's the case then maybe it's too strict of a criteria or we should check another way? but yea agreed we shouldn't create/re-up clusters if theyre down.